src/DynamoCore/Models/CustomNodeWorkspaceModel.cs

```diff
@@ -32,7 +33,7 @@ private set
 
         public CustomNodeWorkspaceModel(
             string name, string category, string description, double x, double y, Guid customNodeId,
-            NodeFactory factory, string fileName="")
+            NodeFactory factory, ElementResolver elementResolver, string fileName="")
```

Comment:
Can `WorkspaceModel` not create `ElementResolver` internally? If it can, then we should definitely remove this, and make a `private readonly ElementResolver elementResolver` that can only be instantiated in the base `WorkspaceModel`. Derived workspace classes can use `WorkspaceModel.ElementResolver` internal property to access it.